### PR TITLE
Create a series of brute force region searches

### PR DESCRIPTION
**Pull request by jeremykubica**
Adds a bunch of proof of concept pointing searches:
- Search a grid of points in barycentric space
- Search a grid of points centered on the barycenter, but pruning samples inside a minimum distance or outside a maximum distance
- Project the geocentric viewing angles to 3-d points with an estimated distance and use those directly as search points
- Project the geocentric viewing angles to 3-d points with an estimated distance and use those to create bounds for a grid search.

Also makes some small extensions to the PointingTables interface such as: `__len__()` and a function to prune rows outside a give time range.

### CODE REVIEW

File: src/pointings_search/grid_iterators.py [hunk 1]
```diff
@@ -0,0 +1,104 @@
...
+
+    def __init__(self, coord_start, coord_end, num_steps, *args, **kwargs):
+        super().__init__(*args, **kwargs)
+
+        self.num_dim = len(coord_start)
+        if self.num_dim != len(coord_end) or self.num_dim != len(num_steps):
+            raise ValueError("Incompatible dimensions.")
+
+        # Create a single product iterator over all the dimensions.
+        self.iter = itertools.product(
```
Comment (wilsonbb) [inline src/pointings_search/grid_iterators.py:32]: The itertools.product functionality is neat!

File: src/pointings_search/search_algorithms.py [hunk 2]
```diff
@@ -0,0 +1,175 @@
...
+from astropy.coordinates import CartesianRepresentation, SkyCoord, get_body_barycentric
+
+from pointings_search.grid_iterators import DistanceGridIterator, GridIterator
+from pointings_search.pointing_table import PointingTable
+
+
+def exhaustive_search(data, options, min_images=5, max_results=100):
+    """Perform a linear search over the options defined by a BaseSearchIterator.
+
+    Current approach is turly brute force and makes no attempt at early pruning.
```
Comment (wilsonbb) [inline src/pointings_search/search_algorithms.py]: s/turly/truly

**Review by wilsonbb — CHANGES_REQUESTED** [review]
LGTM, just a nit on a typo

File: src/pointings_search/search_algorithms.py (hunk 2 above)
Comment (jeremykubica) [reply src/pointings_search/search_algorithms.py]: Fixed.

**Review by wilsonbb — APPROVED** [review]
LGTM, the extra documentation seems helpful!

**Pull request merged**